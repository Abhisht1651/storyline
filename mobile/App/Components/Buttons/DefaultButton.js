// @flow
import React from 'react'
import PropTypes from 'prop-types'
import { View, Text, TouchableWithoutFeedback, StyleSheet } from 'react-native'
// Styles
import styles from '../Styles/ButtonStyles/DefaultButtonStyles'

// Import this file from (App/Components/Buttons)
// and Declare where you need to
// implement it by using
// <DefaultButton title="" onPress={}/>

type Props = {
  title: string,
  disabled: boolean,
  outline: boolean,
  onPress: PropTypes.func,
  style: StyleSheet
}

export const DefaultButton = (props: Props) => {
  const { title, disabled, outline, onPress, style } = props
  return (
    <View
      style={[
        disabled
          ? styles.disableddefaultbutton
          : [outline ? styles.defaultoutlinebutton : styles.defaultbutton],
        style
      ]}>
      <TouchableWithoutFeedback disabled={disabled} onPress={onPress}>
        <View style={styles.text}>
          <Text
            style={[
              disabled
                ? styles.disableddefaulttext
                : [outline ? styles.defaultoutlinetext : styles.defaulttext],
              style
            ]}>
            {title}
          </Text>
        </View>
      </TouchableWithoutFeedback>
    </View>
  )
}

DefaultButton.defaultProps = {
  title: 'Button',
  disabled: false,
  outline: false,
  onPress: () => {},
  style: {}
}
