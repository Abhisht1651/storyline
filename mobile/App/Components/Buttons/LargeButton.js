// @flow
import React from 'react'
import { View, Text, TouchableWithoutFeedback, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'
// Styles
import styles from '../Styles/ButtonStyles/LargeButtonStyles'

// Import this file from (App/Components/Buttons)
// and Declare where you need to
// implement it by using
// <LargeButton title="" onPress={}/>

type Props = {
  title: string,
  disabled: boolean,
  onPress: PropTypes.func,
  style: StyleSheet
}

export const LargeButton = (props: Props) => {
  const { title, disabled, onPress, style } = props
  return (
    <View style={[disabled ? styles.disabledlargebutton : styles.largebutton, style]}>
      <TouchableWithoutFeedback disabled={disabled} onPress={onPress}>
        <View style={styles.text}>
          <Text style={[disabled ? styles.disabledlargetext : styles.largetext, style]}>
            {title}
          </Text>
        </View>
      </TouchableWithoutFeedback>
    </View>
  )
}

LargeButton.defaultProps = {
  title: 'Button',
  disabled: false,
  onPress: () => {},
  style: {}
}
