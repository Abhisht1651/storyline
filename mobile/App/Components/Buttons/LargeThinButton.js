// @flow
import React from 'react'
import { View, Text, TouchableWithoutFeedback, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'
// Styles
import styles from '../Styles/ButtonStyles/LargeThinButtonStyles'

// Import this file from (App/Components/Buttons)
// and Declare where you need to
// implement it by using
// <LargeThinButton title="" onPress={}/>

type Props = {
  title: string,
  disabled: boolean,
  outline: boolean,
  other: boolean,
  onPress: PropTypes.func,
  style: StyleSheet
}

export const LargeThinButton = (props: Props) => {
  const { title, disabled, outline, onPress, style, other } = props
  return (
    <View
      style={[
        disabled
          ? styles.disabledlargethinbutton
          : [
              outline
                ? styles.outlinelargethinbutton
                : [other ? styles.otherlargethinbutton : styles.largethinbutton]
            ],
        style
      ]}>
      <TouchableWithoutFeedback disabled={disabled} onPress={onPress}>
        <View style={styles.text}>
          <Text
            style={[
              disabled
                ? styles.disabledlargethintext
                : [
                    outline
                      ? styles.outlinelargethintext
                      : [other ? styles.otherlargethintext : styles.largethintext]
                  ],
              style
            ]}>
            {title}
          </Text>
        </View>
      </TouchableWithoutFeedback>
    </View>
  )
}

LargeThinButton.defaultProps = {
  title: 'Button',
  disabled: false,
  outline: false,
  other: false,
  onPress: () => {},
  style: {}
}
