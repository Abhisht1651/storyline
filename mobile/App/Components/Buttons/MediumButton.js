// @flow
import React from 'react'
import { View, Text, TouchableWithoutFeedback, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'
// Styles
import styles from '../Styles/ButtonStyles/MediumButtonStyles'

// Import this file from (App/Components/Buttons)
// and Declare where you need to
// implement it by using
// <MediumButton title="" onPress={}/>

type Props = {
  title: string,
  outline: boolean,
  disabled: boolean,
  onPress: PropTypes.func,
  style: StyleSheet
}

export const MediumButton = (props: Props) => {
  const { title, disabled, outline, onPress, style } = props
  return (
    <View
      style={[
        outline
          ? styles.outlinemediumbutton
          : [disabled ? styles.disabledmediumbutton : styles.mediumbutton],
        style
      ]}>
      <TouchableWithoutFeedback onPress={onPress} disabled={disabled}>
        <View style={styles.text}>
          <Text
            style={[
              outline
                ? styles.outlinemediumtext
                : [disabled ? styles.disabledmediumtext : styles.mediumtext],
              style
            ]}>
            {title}
          </Text>
        </View>
      </TouchableWithoutFeedback>
    </View>
  )
}

MediumButton.defaultButton = {
  title: 'Button',
  outline: false,
  disabled: false,
  onPress: () => {},
  style: {}
}
