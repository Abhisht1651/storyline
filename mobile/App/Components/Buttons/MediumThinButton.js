// @flow
import React from 'react'
import { View, Text, TouchableWithoutFeedback, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'
// Styles
import styles from '../Styles/ButtonStyles/MediumThinButtonStyles'

// Import this file from (App/Components/Buttons)
// and Declare where you need to
// implement it by using
// <MediumThinButton title="" onPress={}/>

type Props = {
  title: string,
  disabled: boolean,
  onPress: PropTypes.func,
  style: StyleSheet
}

export const MediumThinButton = (props: Props) => {
  const { title, disabled, onPress, style } = props
  return (
    <View style={[disabled ? styles.disabledmediumthinbutton : styles.mediumthinbutton, style]}>
      <TouchableWithoutFeedback disabled={disabled} onPress={onPress}>
        <View style={styles.text}>
          <Text style={[disabled ? styles.disabledmediumthintext : styles.mediumthintext, style]}>
            {title}
          </Text>
        </View>
      </TouchableWithoutFeedback>
    </View>
  )
}

MediumThinButton.defaultProps = {
  title: 'Button',
  disabled: false,
  onPress: () => {},
  style: {}
}
