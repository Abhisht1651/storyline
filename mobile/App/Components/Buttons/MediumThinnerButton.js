// @flow
import React from 'react'
import { View, Text, TouchableWithoutFeedback, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'
// Styles
import styles from '../Styles/ButtonStyles/MediumThinnerButtonStyles'

// Import this file from (App/Components/Buttons)
// and Declare where you need to
// implement it by using
// <MediumThinnerButton title="" onPress={}/>

type Props = {
  title: string,
  outline: boolean,
  onPress: PropTypes.func,
  style: StyleSheet
}

export const MediumThinnerButton = (props: Props) => {
  const { title, outline, onPress, style } = props
  return (
    <View style={[outline ? styles.outlinemediumthinnerbutton : styles.mediumthinnerbutton, style]}>
      <TouchableWithoutFeedback onPress={onPress}>
        <View style={styles.text}>
          <Text
            style={[outline ? styles.outlinemediumthinnertext : styles.mediumthinnertext, style]}>
            {title}
          </Text>
        </View>
      </TouchableWithoutFeedback>
    </View>
  )
}

MediumThinnerButton.defaultProps = {
  title: 'Button',
  outline: false,
  onPress: () => {},
  style: {}
}
