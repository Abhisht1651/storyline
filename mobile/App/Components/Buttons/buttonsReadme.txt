Details for Buttons Component.

Import it from (mobile/App/Components/Buttons) and declare it where you needs to implement it by <"Button-type" {...props}/>

Common Props: title(string), onPress(function)

Other states/versions of each button can be attained by passing suitable props in it.(Mentioned below)

"Button-type":
LargeButton: {states: disabled(boolean)}
LargeThinButton: {states: disabled(boolean), outline(boolean), other(boolean)}
MediumButton: {states: outline(boolean)}
MediumThinButton: {states: disabled(boolean)}
MediumThinnerButton: {states: outline(boolean)}
SmallButton: {no other states}
XSButton: {states: disabled(boolean), outline(boolean)}
X2SButton: {states: disabled(boolean), outline(boolean), float(boolean)}
X3SButton: {states: outline(boolean)}
X4SButton: {states: disabled(boolean), outline(boolean)}
DefaultButton: {states: disabled(boolean), outline(boolean)}


Note: Default state of Buttons is 'Solid/Fill'.
