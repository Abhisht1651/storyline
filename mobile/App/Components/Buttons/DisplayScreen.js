import React from 'react'
import { View, StyleSheet } from 'react-native'

import { DefaultButton } from './DefaultButton'
import { LargeButton } from './LargeButton'
import { LargeThinButton } from './LargeThinButton'
import { MediumButton } from './MediumButton'

// import { MediumThinButton } from './MediumThinButton'
// import { MediumThinnerButton } from './MediumThinnerButton'
// import { SmallButton } from './SmallButton'

// import { X2SButton } from './X2SButton'
// import { X3SButton } from './X3SButton'
// import { X4SButton } from './X4SButton'
// import { XSButton } from './XSButton'

// import styles from '../Styles/DisplayScreenStyle'

const DisplayScreen = () => {
  return (
    <View style={styles.container}>
      {/* Default buttons */}
      <View style={styles.default}>
        <DefaultButton title="Accept" />
        <DefaultButton title="Add" outline />
        <DefaultButton title="Added" outline disabled />
      </View>

      {/* Large buttons */}
      <View style={styles.large}>
        <LargeButton title="Continue" />
        <LargeButton title="Continue" disabled />
      </View>

      {/* Large thin buttons */}
      <View style={styles.large}>
        <LargeThinButton title="Post" />
        <LargeThinButton title="Post" disabled />
        <LargeThinButton title="Vote" outline />
        <LargeThinButton title="Share" other />
      </View>

      {/* Medium buttons */}
      <View style={styles.large}>
        <MediumButton title="Get Started" />
        <MediumButton title="Log In" outline />
      </View>

      {/* Medium thin buttons */}
      {/* <View style={styles.large}>
        <MediumThinButton title="Save" onPress={() => alert("Medium Thin Pressed!")}/>
        <MediumThinButton title="Save" disabled/>
      </View> */}

      {/* Medium thinner buttons */}
      {/* <View style={styles.large}>
        <MediumThinnerButton title="Media" outline onPress={() => alert("Medium Thinner Pressed!")}/>
      </View> */}

      {/* Small buttons */}
      {/* <View style={styles.large}>
        <SmallButton title="View Post" onPress={() => alert("Small Pressed!")}/>
      </View> */}

      {/* XS buttons */}
      {/* <View style={styles.large}>
        <XSButton title="Storyline" outline onPress={() => alert("XS Pressed!")}/>
        <XSButton title="Storyline" disabled />
      </View> */}

      {/* X2S buttons */}
      {/* <View style={styles.default}>
        <X2SButton title="New Updates" float onPress={() => alert("X2S Pressed!")}/>
        <X2SButton title="Following" disabled />
      </View> */}

      {/* X3S buttons */}
      {/* <View style={styles.default}>
        <X3SButton title="Edit Profile" outline onPress={() => alert("X3S Pressed!")}/>
      </View> */}

      {/* X4S buttons */}
      {/* <View style={styles.default}>
        <X4SButton title="Follow" outline onPress={() => alert("X4S Pressed!")}/>
        <X4SButton title="Following" disabled />
      </View> */}
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    // alignItems: 'center',
    backgroundColor: '#FFF'
  },
  default: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    padding: 20
  },
  large: {
    padding: 20,
    alignItems: 'center'
  }
})

export default DisplayScreen
