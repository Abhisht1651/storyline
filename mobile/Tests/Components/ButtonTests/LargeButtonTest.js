import { TouchableWithoutFeedback } from 'react-native'
import React from 'react'
import { shallow } from 'enzyme'

import { LargeButton } from '../../../App/Components/Buttons/LargeButton'

// Test for LargeButton (snapshot and props)

describe('LargeButton', () => {
  it('LargeButton component renders correctly', () => {
    const tree = shallow(<LargeButton />)
    expect(tree).toMatchSnapshot()
  })

  it('LargeButton disabled', () => {
    const wrapper = shallow(<LargeButton disabled />).props()
    expect(wrapper.children.props.disabled).toEqual(true)
  })

  it('LargeButton onPress', () => {
    const onPressed = jest.fn()
    const wrapper = shallow(<LargeButton onPress={onPressed} />)
    wrapper.find(TouchableWithoutFeedback).props().onPress()
    expect(onPressed.mock.calls.length).toBe(1)
  })

  it('LargeButton disabled onPress', () => {
    const onPressed = jest.fn()
    const wrapper = shallow(<LargeButton disabled onPress={onPressed} />)
    wrapper.find(TouchableWithoutFeedback).simulate('click')
    expect(onPressed.mock.calls.length).toBe(0)
  })
})
