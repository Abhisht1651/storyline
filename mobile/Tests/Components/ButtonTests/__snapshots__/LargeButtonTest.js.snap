// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LargeButton LargeButton component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#7B46E4",
        "borderRadius": 78,
        "height": 120,
        "width": 650,
      },
      Object {},
    ]
  }
>
  <TouchableWithoutFeedback
    disabled={false}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "BananaGrotesk-Bold",
              "fontSize": 32,
              "letterSpacing": -0.18,
              "lineHeight": 36,
            },
            Object {},
          ]
        }
      >
        Button
      </Text>
    </View>
  </TouchableWithoutFeedback>
</View>
`;
